idea/idea-frontend-fir/src/org/jetbrains/kotlin/idea/frontend/api/fir/components/KtFirSmartcastProvider.kt

```diff
@@ -50,4 +54,22 @@ internal class KtFirSmartcastProvider(
             }?.let(::add)
         }
     }
+
+    override fun getStableNullability(expression: KtExpression): StableNullability {
+        val fir = expression.getOrBuildFir(analysisSession.firResolveState)
+        // TODO: Check stability of smartcast (pending PR 4382)
+        return when (fir) {
+            is FirExpressionWithSmartcastToNull -> StableNullability.NULL
+            is FirExpressionWithSmartcast -> {
+                with(analysisSession.rootModuleSession.typeContext) {
```

Comment:
Could you please specify semantics of StableNullability/getStableNullability by adding kdocs to `KtSmartCastProviderMixIn/getStableNullability` and `StableNullability` enum?
I do not quite understand what it StableNullability should mean and why we only checking it by smartcasts?